performance/src/main/java/org/mule/test/PluginClassloaderCreationBenchmark.java

```diff
@@ -0,0 +1,200 @@
+  private static final BundleDescriptor PLUGIN1_BUNDLE_DESCRIPTOR =
+      new BundleDescriptor.Builder()
+          .setGroupId(GROUP_ID).setArtifactId(PLUGIN_ID1)
+          .setVersion("1.0").setClassifier("mule-plugin").build();
+  private static final BundleDescriptor PLUGIN2_BUNDLE_DESCRIPTOR =
+      new BundleDescriptor.Builder()
+          .setGroupId(GROUP_ID).setArtifactId(PLUGIN_ID2)
+          .setVersion("1.0").setClassifier("mule-plugin").build();
+
+  private final ArtifactPluginDescriptor plugin1Descriptor = new ArtifactPluginDescriptor(PLUGIN_ID1);
+  private final ArtifactPluginDescriptor plugin2Descriptor = new ArtifactPluginDescriptor(PLUGIN_ID2);
+  private final ArtifactPluginDescriptor plugin2DescriptorWithPrivilege = new ArtifactPluginDescriptor(PLUGIN_ID2);
+
+  private final ModuleRepository moduleRepository = mock(ModuleRepository.class);
+  private final DefaultNativeLibraryFinderFactory nativeLibraryFinderFactory = new DefaultNativeLibraryFinderFactory();
+  private final String customDomainName = "custom-domain";
+  private final String applicationName = "app";
+  private final String pluginPackage = "plugin-package";
+  private final String plugin2ExportedPackage = "plugin2-package";
+  private final String package1Name = "module&plugin-package";
+  private final String package2Name = "org.mule.sdk.api.package";
+  private MuleDeployableArtifactClassLoader applicationClassLoader;
+  private ClassLoaderModel pluginDependantClassLoaderModel;
+  private ClassLoaderModel pluginExportingPackageClassLoaderModel;
+  private ClassLoaderModel plugin2ExportingPackageClassLoaderModel;
+  private List<MuleModule> privilegeMuleModuleSingletonList;
+  private ClassLoaderModel pluginDependantWithLocalPackageClassLoaderModel;
+  private List<MuleModule> muleModuleSingletonList;
+  private ClassLoaderModel pluginWithLocalPackageClassLoaderModel;
+
+  @Setup
+  public void setup() throws IOException {
+    TemporaryFolder temporaryFolder = new TemporaryFolder();
+    artifactLocation = new TemporaryFolder();
+    temporaryFolder.create();
+    artifactLocation.create();
+    muleHomeFolder = temporaryFolder.getRoot();
+    System.setProperty(MULE_HOME_DIRECTORY_PROPERTY, temporaryFolder.getRoot().getAbsolutePath());
+
+    artifactClassLoaderResolver = spy(new DefaultArtifactClassLoaderResolver(moduleRepository, nativeLibraryFinderFactory));
+
+    plugin1Descriptor.setBundleDescriptor(PLUGIN1_BUNDLE_DESCRIPTOR);
+    plugin2Descriptor.setBundleDescriptor(PLUGIN2_BUNDLE_DESCRIPTOR);
+
+    customDomainDescriptor = getTestDomainDescriptor(customDomainName);
+    customDomainDescriptor.setRootFolder(createDomainDir(MULE_DOMAIN_FOLDER, customDomainName));
+
+    MuleModule privilegedModule = mock(MuleModule.class);
+    when(privilegedModule.getPrivilegedArtifacts()).thenReturn(singleton(PLUGIN_ARTIFACT_ID1));
+    when(privilegedModule.getPrivilegedExportedPackages()).thenReturn(singleton(PRIVILEGED_PACKAGE));
```

Comment:
beware that calls to mocks may have significant overhead relative to regular calls and distort the microbenchmark